lib/active_merchant/billing/gateways/credorax.rb

```diff
@@ -281,7 +314,7 @@ def add_3d_secure(post, options)
           browser_info = three_ds_2_options[:browser_info]
           post[:'3ds_initiate'] = options[:three_ds_initiate] || '01'
           post[:'3ds_purchasedate'] = Time.now.utc.strftime('%Y%m%d%I%M%S')
-          post[:'3ds_channel'] = '02'
+          options[:stored_credential][:initiator] == 'merchant' ? post[:'3ds_channel'] = '03' : post[:'3ds_channel'] = '02'
```

Comment:
This makes sense, but written this way has the potential to run into `undefined method '[]' for nil:NilClass` error if there is no `[:stored_credential]` hash included in the options. Take a look at ruby's `.dig` method for a safer way to check for `[:initiator]`.

Reply:
Thanks for the tip. The `.dig` method is now being used to check for `[:initiator]`